src/plugins/data/common/index_patterns/errors.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FieldSpec } from './types';
+
+export class FieldTypeUnknownError extends Error {
```

Comment:
Should this be extending `KbnError` instead? (I don't know the right answer to this ATM, but feels like `KbnError` is something we should determine what to do with eventually)

Reply:
Added this to Tuesday's team discussion

Reply:
Result from team discussion - there's no current need to inherit from `KbnError` and if that changes in the future it will be easy to address. `KbnError` was mostly useful for internet explorer support.